### DISCUSSION TITLE
Access x-data from outside Alpine

### DISCUSSION THREAD

**Discussion by florian-sun** [alpinejs/alpine#4425] [1. Help] [answered]
How can I access x-data from outside Alpine?
Something like this:

```
<body>   
  <div x-data="{count: 0}">
     <div some_attribute = xData.count ??? > ... </div>
  </div>

  <script> 
   ...
    let count = Alpine.xData.count  ???
  </script>
</body>
```

**Comment by ekwoka** [comment]
Generally, you shouldn't. For what purpose do you have the need for this?

There can be multiple ways to expose data to other code, and the best way to do that is informed by what exactly the goal is.

Do you have more of an example that demonstrates the need? This code here does not demonstrate a need.

**Comment by florian-sun** [comment]
Sure. I'm trying to use Alpine together with htmx. I want to pass x-data back to server using the htmx attribute hx-vals (https://htmx.org/attributes/hx-vals/)

**Reply by ekwoka** [reply, marked as answer]
Then you can do `:hx-vals="JSON.stringify($data)"`

It was doing exactly that that led me to need to make this PR to htmx https://github.com/bigskysoftware/htmx/pull/2799

**Comment by florian-sun** [comment]
OK, thank you
